pkg/integration/tests/worktree/bare_repo_worktree_config.go

```diff
@@ -0,0 +1,91 @@
+package worktree
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+// This case is identical to dotfile_bare_repo.go, except
+// that it invokes lazygit with $GIT_DIR set but not
+// $GIT_WORK_TREE. Instead, the repo uses the core.worktree
+// config to identify the main worktre.
+
+var BareRepoWorktreeConfig = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Open lazygit in the worktree of a vcsh-style bare repo and add a file and commit",
+	ExtraCmdArgs: []string{"--git-dir={{.actualPath}}/.bare"},
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
```

Comment:
Can we go:
```go
	SetupConfig: func(config *config.AppConfig) {
		config.UserConfig.Gui.ShowFileTree = false
	},
```

And then later on go:
```go
			Lines(
				Contains("M a/b/c/blah"),  // shows as modified
			).
```

Reason being it's a little harder to make sense of the assertion when we're including both lines due to it being shown as a tree.

Same with the other tests

Reply:
Sounds great. To be clear, I'm still new enough to lazygit that I had no idea that was even possible, so this sort of tweak is super helpful.

Reply:
I regularly switch between flat and tree mode from within lazygit (using the backtick key). I find that both forms have different strengths

Reply:
me: watching the integration tests run, and being all "OOOH, I should go find THAT feature..." 😂